app/helpers/record_helper.rb

```diff
@@ -4,6 +4,31 @@ def is_number?(object)
     true if Float(object) rescue false
   end
 
+  def has_associated_granule?(record)
+    record.associated_granule_value.present? && is_number?(record.associated_granule_value) ? true : false
```

Comment:
record.associated_granule_value.present? && is_number?(record.associated_granule_value) will evaluate to true or false, won't it?